### PR TITLE
feat: add handler name to exceptions in handler validation.

### PR DESCRIPTION
**Pull request by rafalkrupinski**
Related: #

## Description

- Add handler name to exceptions raise by _validate_handler_function.

## Closes
 
addresses #3548

### CODE REVIEW

File: litestar/handlers/base.py [hunk 1]
```diff
@@ -254,6 +254,16 @@ def handler_name(self) -> str:
...
 
+    @property
+    def handler_fullname(self) -> str:
+        """Get the qualified name of the handler function.
+
+        Returns:
+            Qualified name of the handler function
+        """
+        handler_fn = unwrap_partial(self.fn)
+        return handler_fn.__module__ + "." + handler_fn.__qualname__
```
Comment (provinzkraut) [inline litestar/handlers/base.py:265]: I think the logic here should be 

```suggestion
        return self.name or self.handler_name
```
so custom handler names show up as well, which means the handler name shown in the exception matches e.g. the one you'd put into `route_reverse`.

File: litestar/handlers/http_handlers/decorators.py [hunk 2]
```diff
@@ -596,7 +596,9 @@ def _validate_handler_function(self) -> None:
             or is_class_and_subclass(return_annotation, File)
             or is_class_and_subclass(return_annotation, ASGIFileResponse)
         ):
-            raise ImproperlyConfiguredException("A response to a head request should not have a body")
+            raise ImproperlyConfiguredException(
+                f"A route handler function '{self.handler_fullname}' that handles HEAD request should not have a body"
```
Comment (provinzkraut) [inline litestar/handlers/http_handlers/decorators.py:600]:
```suggestion
                f"{self.handler_fullname}: Handlers for 'HEAD' requests must not return a value. Either return 'None' or a response type without a body."
```

File: litestar/handlers/websocket_handlers/route_handler.py [hunk 3]
```diff
@@ -85,17 +85,25 @@ def _validate_handler_function(self) -> None:
         super()._validate_handler_function()
 
         if not self.parsed_fn_signature.return_type.is_subclass_of(NoneType):
-            raise ImproperlyConfiguredException("Websocket handler functions should return 'None'")
+            raise ImproperlyConfiguredException(
+                f"Websocket handler function '{self.handler_fullname}' should return 'None'"
```
Comment (provinzkraut) [inline litestar/handlers/websocket_handlers/route_handler.py:89]:
```suggestion
                f"{self.handler_fullname}: WebSocket handlers must return 'None'"
```

File: litestar/handlers/websocket_handlers/route_handler.py [hunk 4]
```diff
@@ -85,17 +85,25 @@ def _validate_handler_function(self) -> None:
...
+            )
 
         if "socket" not in self.parsed_fn_signature.parameters:
-            raise ImproperlyConfiguredException("Websocket handlers must set a 'socket' kwarg")
+            raise ImproperlyConfiguredException(
+                f"Websocket handler '{self.handler_fullname}' must set a 'socket' kwarg"
```
Comment (provinzkraut) [inline litestar/handlers/websocket_handlers/route_handler.py:94]:
```suggestion
                f"{self.handler_fullname}: WebSocket handlers must define a 'socket' parameter"
```

File: litestar/handlers/websocket_handlers/route_handler.py [hunk 5]
```diff
@@ -85,17 +85,25 @@ def _validate_handler_function(self) -> None:
...
+            )
 
         for param in ("request", "body", "data"):
             if param in self.parsed_fn_signature.parameters:
-                raise ImproperlyConfiguredException(f"The {param} kwarg is not supported with websocket handlers")
+                raise ImproperlyConfiguredException(
+                    f"{self.handler_fullname}: the {param} kwarg is not supported with websocket handlers"
```
Comment (provinzkraut) [inline litestar/handlers/websocket_handlers/route_handler.py:100]:
```suggestion
                    f"{self.handler_fullname}: The {param} kwarg is not supported with websocket handlers"
```

File: litestar/handlers/websocket_handlers/route_handler.py [hunk 6]
```diff
@@ -85,17 +85,25 @@ def _validate_handler_function(self) -> None:
...
+                )
 
         if not is_async_callable(self.fn):
-            raise ImproperlyConfiguredException("Functions decorated with 'websocket' must be async functions")
+            raise ImproperlyConfiguredException(
+                f"Function '{self.handler_fullname}' decorated with 'websocket' must be async functions"
```
Comment (provinzkraut) [inline litestar/handlers/websocket_handlers/route_handler.py:105]:
```suggestion
                f"{self.handler_fullname}: WebSocket handler functions must be asynchronous"
```

**Review by provinzkraut — CHANGES_REQUESTED** [review]
Thanks @rafalkrupinski!

I've left a few suggestions for consistency

File: litestar/handlers/base.py (hunk 1 above)
Comment (rafalkrupinski) [reply litestar/handlers/base.py:265]: But this way we're loosing package and class name...

Comment (provinzkraut) [reply litestar/handlers/base.py:265]: How about just using `str(handler.fn)`? I'm hesitant to add yet another handler identifier that does something slightly different; We already have `handler_id`, `name` and `handler_name`.

Maybe we could change `handler_name` in v3 to use the fully qualified name instead.

Comment (rafalkrupinski) [reply litestar/handlers/base.py:265]: Actually `str(self)` does exactly that (module+qualname)

**Review by provinzkraut — APPROVED** [review]

**Pull request merged**